js/join/ui/session/refilter.js

```diff
@@ -109,10 +112,16 @@ const populate = (parent, seshData, statusManager, previewManager, notifySel) =>
     }
   }
 
-  container.innerHTML = "";
-  nodes.forEach((node) => container.appendChild(node));
+  sessionRows.innerHTML = "";
+  nodes.forEach((node) => sessionRows.appendChild(node));
+
+  Array.from(sessionRows.querySelectorAll(".session-row")).forEach((row) => {
+      row.addEventListener("click", () => {
+        row.querySelector(".session-option").checked = true;
+        row.querySelector(".session-option").dispatchEvent(new Event("change"));
```

Comment:
Where is this phony event necessary?  Seems indicative of some other error on our part.

Reply:
> Because the `<input>` element that gets checked (once we select a session row in the table) is hidden (whereas before, it was not hidden), when I check it, the change event does not actually fire

But the old code operated on this same principle, didn't it?  Line 59 isn't a new line of code, and reacted to a hidden checkbox being changed.  I think you or I have to be misunderstanding something here.

I don't doubt that this change was intentional, but it's not code that fits a straightforward and correct way of accomplishing what it does.  We should almost never have to fake events.  Either we should respond to an actual event that's firing when we set `checked = true` (which is confusing to me, because I'm not understanding how that could *not* trigger `change`), or we should remove the `change` listener and run its code here, instead of plugging in a fake event.

Your explanation for the behavior that you saw that caused you to make this correction is that the element was not previously hidden, but there's some gap in the understanding (or terms being used) here, because the `input` *is* hidden (by my definition of that word) on the old page:

![image](https://user-images.githubusercontent.com/912890/151041023-aab37a8b-3555-49dc-ac94-cd50a1155d17.png)

Reply:
Resolved together!